src/pages/settings/Profile/ProfilePage.js

```diff
@@ -67,55 +66,28 @@ class ProfilePage extends Component {
+        this.avatar = {uri: lodashGet(this.props.currentUserPersonalDetails, 'avatar', ReportUtils.getDefaultAvatar(this.props.currentUserPersonalDetails.login))};
+        this.pronouns = props.currentUserPersonalDetails.pronouns;
         this.state = {
-            firstName: this.props.currentUserPersonalDetails.firstName,
-            hasFirstNameError: false,
-            lastName: this.props.currentUserPersonalDetails.lastName,
-            hasLastNameError: false,
-            pronouns: this.props.currentUserPersonalDetails.pronouns,
-            hasPronounError: false,
-            hasSelfSelectedPronouns: !_.isEmpty(this.props.currentUserPersonalDetails.pronouns) && !this.props.currentUserPersonalDetails.pronouns.startsWith(CONST.PRONOUNS.PREFIX),
-            selectedTimezone: lodashGet(this.props.currentUserPersonalDetails.timezone, 'selected', CONST.DEFAULT_TIME_ZONE.selected),
-            isAutomaticTimezone: lodashGet(this.props.currentUserPersonalDetails.timezone, 'automatic', CONST.DEFAULT_TIME_ZONE.automatic),
-            logins: this.getLogins(props.loginList),
-            avatar: {uri: lodashGet(this.props.currentUserPersonalDetails, 'avatar', ReportUtils.getDefaultAvatar(this.props.currentUserPersonalDetails.login))},
+            isAutomaticTimezone: lodashGet(props.currentUserPersonalDetails.timezone, 'automatic', CONST.DEFAULT_TIME_ZONE.automatic),
+            hasSelfSelectedPronouns: !_.isEmpty(props.currentUserPersonalDetails.pronouns) && !props.currentUserPersonalDetails.pronouns.startsWith(CONST.PRONOUNS.PREFIX),
             isAvatarChanged: false,
         };
 
         this.getLogins = this.getLogins.bind(this);
-        this.setAutomaticTimezone = this.setAutomaticTimezone.bind(this);
+        this.validate = this.validate.bind(this);
         this.updatePersonalDetails = this.updatePersonalDetails.bind(this);
-        this.validateInputs = this.validateInputs.bind(this);
         this.updateAvatar = this.updateAvatar.bind(this);
     }
 
     componentDidUpdate(prevProps) {
-        let stateToUpdate = {};
-
         // Recalculate logins if loginList has changed
-        if (this.props.loginList !== prevProps.loginList) {
-            stateToUpdate = {...stateToUpdate, logins: this.getLogins(this.props.loginList)};
-        }
-
-        if (_.isEmpty(stateToUpdate)) {
+        if (this.props.loginList === prevProps.loginList) {
             return;
         }
-
-        // eslint-disable-next-line react/no-did-update-set-state
-        this.setState(stateToUpdate);
-    }
-
-    /**
-     * Set the form to use automatic timezone
-     *
-     * @param {Boolean} isAutomaticTimezone
-     */
-    setAutomaticTimezone(isAutomaticTimezone) {
-        this.setState(({selectedTimezone}) => ({
-            isAutomaticTimezone,
-            selectedTimezone: isAutomaticTimezone ? moment.tz.guess() : selectedTimezone,
-        }));
+        this.logins = this.getLogins(this.props.loginList);
```

Comment:
I'm not sure the component will re-render to show the newly added login. When I try to add a phone number, the new phone number only shows up after I trigger a re-render by changing some other value. Maybe this needs to be moved to state.

Reply:
Moved to state.